spriteutils.ts

```diff
@@ -39,6 +39,39 @@ namespace spriteutils {
         return Math.sqrt(((a.x - b.x) ** 2) + ((a.y - b.y) ** 2));
     }
 
+    /**
+     * Returns array of sprites of kind that are within a specified distance.
+     */
+    //% block="get all sprites of kind $kind within $distance pixels from $sprite"
+    //% kind.shadow=spritekind
+    //% distance.defl=50
+    //% sprite.shadow=variables_get
+    //% sprite.defl=mySprite
+    export function getSpritesWithin(kind: number, distance: number, sprite: Sprite) : Sprite[] {
+        let allSprites = sprites.allOfKind(kind);
+        let sortArray : Sprite[] = [];
+        let numItems = 0;
+        allSprites.forEach(function(value: Sprite, index: number) {
+            if (distanceBetween(value, sprite) <= distance) {
+                sortArray.push(value);
+            }
+        })
+
+        sortArray = sortArray.sort((a, b)=>{
+            let distanceA = (distanceBetween(a, sprite));
+            let distanceB = distanceBetween(b, sprite)
+            if (distanceA < distanceB) {
+                return -1;
+            } else if (distanceA > distanceB) {
+                return 1;
+            } else {
+                return 0;
+            }
+        })
```

Comment:
```suggestion
        sortArray = sortArray.sort((a, b) => distanceBetween(a, sprite) - distanceBetween(b, sprite));
```

can just return distanceA - distanceB here right?